debezium-connector-postgres/src/main/java/io/debezium/connector/postgresql/snapshot/NeverSnapshotter.java

```diff
@@ -45,7 +45,7 @@ public boolean shouldSnapshot() {
     }
 
     @Override
-    public Optional<String> buildSnapshotQuery(TableId tableId) {
+    public Optional<String> buildSnapshotQuery(TableId tableId, String snapshotSelectColumns) {
```

Comment:
I think we should pass the list of column names here.

Reply:
Why exactly? Because even if we pass a list of columns, we still would need to convert into a string so that we have our select statement as `Optional.of(String.format("SELECT %s from %s", snapshotSelectColumns, tableId.toDoubleQuotedString())`.

Reply:
Yes, that's true. I just feel passing the separate names allows for more flexibility down the road.

Reply:
That said, I don't think it's causing any failure in the tests. It's related to the schema only. I am not sure what extra implementations are used by connectors using `HistorizedRelationalDatabaseSchema`, which isn't used by Postgres, so after the change, there is a failure, `NullPointerException` somewhere.

Reply:
Do you have a reference to that NPE?

Reply:
All of the test cases are failing in RecordsStreamProducerIT I see. ref: https://github.com/debezium/debezium/pull/2509/checks?check_run_id=3369719027, it's still running.

Reply:
@ani-sha, hum, so this job timed out after 6h, it might have been a temporary issue in the environment. Taking a step back, does this change pass all tests for you if you build locally on your machine?

Reply:
@gunnarmorling The tests for Postgres fail locally too with the change of using `RelationalDatabaseSchema` in `RelationalSnapshotChangeEventSource`. Haven't been able to understand the reason of those NPE failure in depth.